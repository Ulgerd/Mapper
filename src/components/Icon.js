import React from 'react';
import Icons from '../assets/icons.svg';
import styled from 'styled-components'

const StyledPieceImage = styled.svg`
  margin-top: 0.1em;
  justify-self: center;
  cursor: pointer;
  &:hover {
    fill: lightblue;
  }
  `

export default function Icon(props) {
  return (
    <StyledPieceImage
        fill= {props.fill}
        width='1.5em'
        height='1.5em'
        onClick={props.onClick}
      >
        <use xlinkHref={`${Icons}#${props.xlink}`}/>
    </StyledPieceImage>
  )
}
