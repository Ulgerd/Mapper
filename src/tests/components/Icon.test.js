import React from 'react';
import { shallow } from 'enzyme'
import Icon from '../../components/Icon.js';
import renderer from 'react-test-renderer';

describe('Icon component tests',()=>{
    let wrapper

    const props = {
      fill: '#222',
      xlink: 'close'
    };

    beforeEach(()=>{
        wrapper = shallow(Icon(props))
    })

    it('render the DUMB component', () => {
       expect(wrapper.length).toEqual(1)
    });

    it('sets the svg element fill', () => {
      expect(wrapper.find('[fill="#222"]')).toHaveLength(1);
      expect(wrapper.find('[xlinkHref="icons.svg#close"]')).toHaveLength(1);
    });

    it('contains use', () => {
        expect(wrapper.find('use')).toHaveLength(1)
    });

    it('renders correctly', () => {
      const tree = renderer
        .create(<Icon />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
});
