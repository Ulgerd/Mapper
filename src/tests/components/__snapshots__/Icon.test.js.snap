// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component tests renders correctly 1`] = `
<svg
  className="sc-bdVaJa eUzTN"
  height="1.5em"
  width="1.5em"
>
  <use
    xlinkHref="icons.svg#undefined"
  />
</svg>
`;
